src/features_creator/features_creator.py

```diff
@@ -32,21 +35,28 @@ def get_matching_column_names(data, pattern):
         "othercolumn": [5, 6, 7]}
     >>> df = pd.DataFrame(data)
     >>> get_matching_column_names(df, "week_payment")
-        week_payment1  week_payment2  week_payment3
-     0              1              1              1
-     1              2              2              2
-     2              3              3              3
+        ["week_payment1", "week_payment2", "week_payment3"]
 
     """
-    ...
+    if not isinstance(data, pd.DataFrame):
+        raise TypeError("The data variable needs to be a pandas dataframe")
+    if not isinstance(pattern, str):
+        raise TypeError("The pattern variable needs to be a string")
 
+    pattern = f"{pattern}\d+"
```

Comment:
```suggestion
    pattern = fr"{pattern}\d+"
```

I got the warning DeprecationWarning: invalid escape sequence when running this, so just want a bit of future proof: https://stackoverflow.com/questions/50504500/deprecationwarning-invalid-escape-sequence-what-to-use-instead-of-d

Reply:
Aha! I didn't realize you could do both `f` and `r` together.